### DISCUSSION TITLE
Export MDX to Markdown

### DISCUSSION THREAD

**Discussion by ChrisChinchilla** [mdx-js/mdx#1788] [Q&A] [answered]
I know this may seem like a strange question, but I wondered if anyone knew of a way to export MDX content to a markdown file, ideally also rendering any variables with their current value, almost hijacking the HTML generation process and exporting markdown instead.

Basically I need to export the contents of an MDX file in a documentation site to a readme for another GitHub repo…

**Comment by ChrisChinchilla** [comment, marked as answer]
Found this - https://github.com/mdx-js/mdx/discussions/1237?sort=new

**Comment by wooorm** [comment]
Perhaps a good solution is to:
- render it somewhere with something like `React.renderToStaticMarkup` — either do this manually, but alternatively if you’re already doing this somewhere—statically generating HTML or generating an RSS feed—use that and select the generated HTML.
- Turn *that* into markdown:
  ```js
  import {promises as fs} from 'node:fs'
  import {select} from 'hast-util-select'

  const file = await unified()
    .use(rehypeParse)
    // Optional: select the element that contains the generated HTML.
    .use(() => (tree) => select('.main', tree))
    .use(rehypeRemark)
    .use(remarkStringify)
    .process('<!doctype html>…')

  await fs.writeFile('readme.md', file.value)
  ```

**Reply by ChrisChinchilla** [reply]
OK, thanks @wooorm I'll take a look into this and see if it all fits into our current build process as the docs are a kind of curious sub-site in the web app, so the build process is sometimes out of my control.
